### ISSUE TITLE
Infinite loop on decode

### ISSUE THREAD

**Issue by mkamioner** [mafintosh/protocol-buffers#27]
Hello!

First off, thank you so much for your awesome module! I am really enjoying it!

I have the following protobuf file:
```
package com.abchappy;

enum AbcType {
    IGNORE =  0;
    ACK_CONFIRMATION_TOKEN =  1;
}

message AbcAcknowledgeConfirmationToken {
    optional uint64 confirmation_token = 1;
    extensions 1000 to max;
}

message ABC {
    required AbcType type = 9;
    required uint32 api_version = 8;
    optional AbcAcknowledgeConfirmationToken ack_confirmation_token = 1;
    extensions 1000 to max;
}
```
During testing, I performed the following and the module got caught in an **infinite loop!**
```
myProtoBuf.ABC.decode(new Buffer([7, 0, 8, 182, 168, 235, 144, 178, 41]));
```
I need to make sure that if someone accidentaly or maliciously sends something like that to my server, I won't get caught in an infinite loop :-(

Any idea how that can be fixed?

Thanks in advance!

Mike

**Comment by mafintosh** [comment]
Which version of protocol-buffers are you using?
I can't seem to reproduce. If I run the below example

``` js
var protobuf = require('../../')
var messages = protobuf(require('fs').readFileSync('schema.proto'))

messages.ABC.decode(new Buffer([7, 0, 8, 182, 168, 235, 144, 178, 41]));
```

(where schema.proto is your schema) It crashes with the following error

```
/Users/maf/dev/node_modules/protocol-buffers/compile.js:27
  throw new Error('Unknown wire type: '+type)
        ^
Error: Unknown wire type: 7
    at skip (/Users/maf/dev/node_modules/protocol-buffers/compile.js:27:9)
    at Object.decode (eval at <anonymous> (/Users/maf/dev/node_modules/protocol-buffers/node_modules/generate-function/index.js:55:21), <anonymous>:36:16)
    at Object.<anonymous> (/Users/maf/dev/node_modules/protocol-buffers/issues/27/index.js:4:14)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:929:3
```

**Comment by mkamioner** [comment]
I am using 2.4.6

However - there was a typo when I was refactoring the protobuf schema above. Can you try again (it has been udpated since) and see if you get the same error?

**Comment by mkamioner** [comment]
I may have incorrectly refactored in my example.

The following is copy pasted code that fails:
```
'use strict';

var protoStr = 'package com.abchappy;\n' +
    'enum AbcType {\n' +
    '    IGNORE                 =  0;\n'+
    '    ACK_CONFIRMATION_TOKEN =  1;\n'+
    '}\n'+
    'message AbcAcknowledgeConfirmationToken { // 0x01\n' +
    '    optional uint64 confirmation_token = 1;\n'+
    '    extensions 1000 to max;\n'+
    '}\n'+
    'message ABC {\n' +
    '    required AbcType type = 9;\n' +
    '    required uint32 api_version = 8;\n'+
    '    optional AbcAcknowledgeConfirmationToken ack_confirmation_token = 1;\n' +
    '    extensions 1000 to max;\n'+
    '}\n',
    protobuf = require('../'),
    messages = protobuf(protoStr);

console.log('Start');
messages.ABC.decode(new Buffer([8, 182, 168, 235, 144, 178, 41]));
console.log('Done');
```

**Comment by mkamioner** [comment]
I made a pull request to fix it: https://github.com/mafintosh/protocol-buffers/pull/28

**Comment by mafintosh** [comment]
Fixed by #28 which is released in 2.4.7
